File: OnPoolClient/Pool.cs
using System;
using System.Linq;
using System.Security.Authentication.ExtendedProtection;
using OnPoolCommon;
using OnPoolCommon.Models;

namespace OnPoolClient
{
    public class Pool
    {
        private OnMessage onMessage { get; set; }
        public string PoolName { get; set; }

        public Pool(string poolName)
        {
            PoolName = poolName;
        }

        public void ReceiveMessage(Client from, Message message, RespondMessage respond)
        {
            onMessage?.Invoke(from, message, respond);
        }

        public void OnMessage(OnMessage callback)
        {
            onMessage += callback;
        }
    }

    public class Client
    {
        public long Id { get; set; }

        public Client(long clientId)
        {
            Id = clientId;
        }
    }
}

File: OnPoolClientTS/src/pool.ts
import { Message } from "./common/Message";
import { OnMessage, RespondMessage } from "./onPoolClient";

export class Pool {
    private onMessage: OnMessage[] = [];
    public PoolName: string;

    constructor(poolName: string) {
        this.PoolName = poolName;
    }

    public ReceiveMessage(from: Client, message: Message, respond: RespondMessage): void {
        this.invokeMessage(from, message, respond);
    }

    public OnMessage(callback: OnMessage): void {
        this.onMessage.push(callback);
    }

    public invokeMessage(from: Client, message: Message, respond: RespondMessage) {
        for (let i = 0; i < this.onMessage.length; i++) {
            this.onMessage[i](from, message, respond);
        }
    }

}

export class Client {
    public Id: number;

    constructor(clientId: number) {
        this.Id = clientId;
    }
}